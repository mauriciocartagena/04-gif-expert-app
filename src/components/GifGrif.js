import React from 'react'
import PropTypes from 'prop-types';

import { useFectGifs } from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem';

export const GifGrif = ({ category }) => {

    // para renombrar data:images
    const {  data:images, loading } = useFectGifs( category );
    
    return (
        <>
            <h3 className="animate__animated animate__fadeIn" >{ category }</h3>
            {
                loading && <p className="animate__animated animate__flash">Loading...</p>
            }
             <div className="card-grid">  
                {/* <h3>{ count } </h3>
                <button onClick={ ()=> setCount( count+1 ) }></button> */}
                    {
                        images.map(( img )=>{
                            return(
                                <GifGridItem 
                                
                                    key={ img.id }
                                    { ...img }
                                >
                                </GifGridItem>
                            )
                        })  
                    } 

            </div>
        </>
    )
}
GifGrif.propTypes= {

    category:PropTypes.string.isRequired,

}