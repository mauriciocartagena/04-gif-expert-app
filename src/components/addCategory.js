import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ( { setCategories } ) => {

    const [inputValue, setInputValue] = useState('');

    const handleInputChange = ( e ) =>{

        //cambiamos el valor de InputValue
        setInputValue(e.target.value);

    }

    const handleSubmit = (e) =>{

        //evitamos que se haga el reinicio de la pagina al presionar el submit o "Enter "
        e.preventDefault();

        if(inputValue.trim().length > 2 ){

            setCategories( cats => [ inputValue,...cats ]);
            setInputValue('');
        }
         
    }

    return (
        <>
            <form onSubmit={ handleSubmit } >
                <input 
                    type="text"
                    value={ inputValue }
                    onChange={ (e)=> handleInputChange(e) }
                />
            </form>
        </>
    )
}
AddCategory.propTypes = {
    
    setCategories:PropTypes.func.isRequired,
     
}
