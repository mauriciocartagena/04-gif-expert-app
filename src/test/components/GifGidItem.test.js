import '@testing-library/jest-dom';
import React from 'react';
import { shallow } from 'enzyme';


import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas de <GifGridItem />',()=>{

    const data = {

        title:'Hello World',
        url:'https://img2.rtve.es/a/5486015?w=1600&preview=1579261973819.jpg',
    }

    const { title,url } = data;

    const wrapper = shallow(<GifGridItem  

            title = { title }
            url = { url }

        />
    );
    

    test('Debe de mostrar <GifGridItem /> correctaménte', () => {

        
        expect(wrapper).toMatchSnapshot();

    });

    test('Debe de tener párrafo con el title',() => {

        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe( title );

    });

    test('Debe de tener la imagen igual al url y alt de los props',() =>{ 

        const img = wrapper.find('img');
        
        expect( img.prop('src') ).toBe( url );
        expect( img.prop('alt') ).toBe( title );

    });

    test('debe de tener animate__fadeIn ',() =>     { 

        const div = wrapper.find('div');
        // console.log(div.prop('className'));

        const className = div.prop('className');


        expect(className.includes('animate__fadeIn')).toBe(true);

    });

    

});