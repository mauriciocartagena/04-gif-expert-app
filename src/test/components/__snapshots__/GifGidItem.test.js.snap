// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de <GifGridItem /> Debe de mostrar <GifGridItem /> correctam√©nte 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Hello World"
    src="https://img2.rtve.es/a/5486015?w=1600&preview=1579261973819.jpg"
  />
  <p>
     
    Hello World
     
  </p>
</div>
`;
